packages/core/src/modules/job-listing/job-listing-location-link-repository.ts

```diff
@@ -1,40 +1,38 @@
 import { type Database } from "@dotkomonline/db"
-import { type DB } from "@dotkomonline/db/src/db.generated"
-import { type JobListingId } from "@dotkomonline/types"
-import { type Insertable, type Kysely, type Selectable } from "kysely"
-
-type Location = DB["jobListingLocation"]
-type LocationSelect = Selectable<Location>
-type LocationId = LocationSelect["id"]
-
-type Link = DB["jobListingLocationLink"]
-type LinkSelect = Selectable<Link>
-type LinkInsert = Insertable<Link>
-interface LinkDelete {
-  jobListingId: JobListingId
-  locationId: LocationId
-}
+import { type JobListingId, type JobListingLocationId } from "@dotkomonline/types"
+import { type Kysely } from "kysely"
 
 export interface JobListingLocationLinkRepository {
-  add(payload: LinkInsert): Promise<LinkSelect>
-  remove(payload: LinkDelete): Promise<void>
+  add(jobListingId: JobListingId, locationId: JobListingLocationId): Promise<void>
+  remove(jobListingId: JobListingId, locationId: JobListingLocationId): Promise<void>
+  removeByLocationName(jobListingId: JobListingId, locationName: string): Promise<void>
```

Comment:
Instead of doing a lot of ID comparisons and fetching, I added a sql query to remove by name